bindings/src/main/java/io/pravega/storage/hdfs/HDFSStorage.java

```diff
@@ -262,10 +278,30 @@ public void seal(SegmentHandle handle) throws StreamSegmentException {
     public void unseal(SegmentHandle handle) throws StreamSegmentException {
         ensureInitializedAndNotClosed();
         long traceId = LoggerHelpers.traceEnter(log, "seal", handle);
+        Path targetPath = getFilePath(handle.getSegmentName(), this.epoch);
+        Path sealedPath = getSealedFilePath(handle.getSegmentName());
+
         try {
-            FileStatus status = findStatusForSegment(handle.getSegmentName(), true);
-            makeWrite(status);
-            this.fileSystem.rename(status.getPath(), getFilePath(handle.getSegmentName(), this.epoch));
+            makeWrite(sealedPath);
```

Comment:
Should this happen before or after the rename?

Reply:
For the operation seal I'm using rename before making file read-only. To possibly leaving file read only when seal fails without sealing it.
For unseal , I'm using opposite sequence.

Reply:
I was originally planning to not have any rename+makewrite in catch block. In normal circumstances, an already created file should not disappear from file system for rename operation and somehow magically reappear in globStatus. I'm thinking of removing the re-attempted renames  from catch blocks .  I can not think of a real life scenario where that code is going to get executed.

What do you think @tkaitchuck @andreipaduroiu ?

Reply:
Please add a test covering this case.

Reply:
sure.